src/libecalc/common/graph.py

```diff
@@ -27,11 +27,11 @@ def add_node(self, node: TNode) -> Self:
         self.nodes[node.id] = node
         return self
 
-    def add_edge(self, from_id: NodeID, to_id: NodeID) -> Self:
+    def add_edge(self, from_id: NodeID, to_id: NodeID, **kwargs) -> Self:
```

Comment:
docstring? intention behind kwargs ... different properties based on type of edge etc?